praw/objector.py

```diff
@@ -161,6 +162,22 @@ def objectify(
         if "json" in data and "errors" in data["json"]:
             errors = data["json"]["errors"]
             if len(errors) > 0:
+                for error in errors:
+                    if "ratelimit" in [
+                        str(index(error, 0)).lower(),
+                        str(index(error, 2)).lower(),
+                    ]:
+                        amount_search = search(
+                            r"([0-9]{1,2}) (seconds?|minutes?)", error[1]
+                        )
+                        if not amount_search:
+                            break
+                        secs = int(amount_search.group(1))
+                        if "minute" in amount_search.group(2):
+                            secs *= 60
+                        if secs <= self._reddit.ratelimit_seconds:
+                            time.sleep(secs)
```

Comment:
I think that maybe we should do `seconds += 1` because if Reddit says "wait 5 seconds," that might actually mean "wait 5.3 seconds" and if we sleep for exactly 5 seconds, our retry will be too soon. Of course, if we add one to the wait time, we need to compare the "updated" wait time to our configured maximum.

On the other hand, that might not be the case, in which case we shouldn't waste the extra second. I'm wondering what your testing has shown — do you ever retry too soon?

Reply:
I compromised with an algorithim that sleeps the seconds **plus 1/10th of the seconds to sleep or 1 second, whichever is smallest**